boards/arm/nucleo_f103rb/pinmux.c

```diff
@@ -14,15 +14,15 @@
 
 /* pin assignments for NUCLEO-F103RB board */
 static const struct pin_config pinconf[] = {
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi1), okay) && CONFIG_SPI
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi1), okay) && defined(CONFIG_SPI)
```

Comment:
There are more than 250 occurrences in current master of 
`#if DT_NODE_HAS_STATUS(DT_NODELABEL(foox), okay) && defined(CONFIG_FOO)`

Some of them (which current occurrence is part of), will disappear as part of #28999.
But I think this should be addressed globally anyway.

Reply:
@erwango I agree, per this commit message:

https://github.com/zephyrproject-rtos/zephyr/pull/29208/commits/a4befb2c6c7ea1bd9bf6ec5c12fb4bc02cc7ea49

"Prevent `Wundef` warnings from occurring due to missing CONFIG_ symbols.
Note that the pattern fixed here is pervasive across other files in the
`boards/` directory, and it may be worth applying the same changes
to all other cases found with `ack "\) && CONFIG"`."

Unfortunately, I have no time to spend on this effort in the next 1-2 weeks, but feel free to contribute a more global solution if you wish.